### PR TITLE
Added index idx_published_ordering to language table

### PR DESCRIPTION
**Pull request by nadeeshaan**
In this pull request I have added an index for the language table to improve the performance of this query

SELECT *
  FROM #__languages
  WHERE published=1
  ORDER BY ordering ASC

### CODE REVIEW

**Comment by nadeeshaan** [comment]
Testing:
In order to check this commit first you need to have a fresh joomla distribution and then pull this PR before the  installation.If you are testing on an already installed distribution you can manually add the above index to the #__languages table.
You can see this query execution when you access the site(not necessarily the admin panel)

SELECT *
FROM #__languages
WHERE published=1
ORDER BY ordering ASC

Also please examine the EXPLAIN statistics too. Which will give you a better understanding why this index is so necessary

**Comment by alikon** [comment]
please remove the star

File: libraries/joomla/language/helper.php [hunk 1]
```diff
@@ -157,7 +157,8 @@ public static function getLanguages($key = 'default')
 				{
 					$db = JFactory::getDbo();
 					$query = $db->getQuery(true)
-						->select('*')
+						->select('lang_id,lang_code,title,title_native,sef,image,description,metakey,metadesc,
```
Comment (gunjanpatel) [inline libraries/joomla/language/helper.php]: To improve readability of code, please use,

```
<?php

$query = $db->getQuery(true)
->select(
    array(
        'lang_id',
        'lang_code',
        'title',
        'title_native',
        'sef',
        'image',
        'description',
        'metakey',
        'metadesc',

    )
);
```

**Comment by alikon** [comment]
for testing on an installed site run this 1 sql line
`ALTER TABLE #__languages ADD INDEX 'idx_published_ordering' ('published','ordering');`

**Comment by gunjanpatel** [comment]
Closing this as this is Merged in https://github.com/nadeeshaan/JGSoC-SQL-Optimisation/pull/29

**Pull request closed without merging**